### PR TITLE
feat(red): create module

### PR DESCRIPTION
**Pull request by Milo123459**
#### Description

Adds a module for Red
#### Motivation and Context

Closes #

#### Screenshots (if appropriate):
img
#### How Has This Been Tested?

- [ ] I have tested using **MacOS**
- [ ] I have tested using **Linux**
- [x] I have tested using **Windows**

#### Checklist:

- [x] I have updated the documentation accordingly.
- [x] I have updated the tests accordingly.

### CODE REVIEW

**Comment by vladimyr** [comment]
> `fmt even though i already did it`
💯

File: docs/config/README.md [hunk 1]
```diff
@@ -2236,6 +2237,45 @@ python_binary = "python3"
...
+| style\*  |          | Mirrors the value of option `style`  |
+
+\*: This variable can only be used as a part of a style string
+
+### Example
+
+```toml
+# ~/.config/starship.toml
+
+[ruby]
```
Comment (vladimyr) [inline docs/config/README.md:2275]:
```suggestion
[red]
```

File: src/modules/red.rs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                _ => None,
+            })
+            .map_style(|variable| match variable {
+                "style" => Some(Ok(config.style)),
+                _ => None,
+            })
+            .map(|variable| match variable {
+                "version" => context
+                    .exec_cmd("red", &["--version"])
+                    .and_then(|output| parse_red_version(output.stdout.trim()))
```
Comment (vladimyr) [inline src/modules/red.rs:32]: Use `.map` here instead so you can get rid of unnecessary option wrap.

File: src/modules/red.rs [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            log::warn!("Error in module `red`:\n{}", error);
+            return None;
+        }
+    });
+
+    Some(module)
+}
+
+#[allow(clippy::clippy::unnecessary_wraps)]
+fn parse_red_version(red_version: &str) -> Option<String> {
```
Comment (vladimyr) [inline src/modules/red.rs:51]: Return `String` instead.

**Review by vladimyr — CHANGES_REQUESTED** [review]
LGTM, just a couple of small fixes left.
Also, please post hyperfine stats for `red --version` invocation.

File: docs/config/README.md (hunk 1 above)
Comment (Milo123459) [reply docs/config/README.md:2275]: My bad, as per usual...

File: docs/config/README.md [hunk 4]
```diff
@@ -2236,6 +2237,45 @@ python_binary = "python3"
 detect_extensions = []
 ```
 
+## Red
+
+By default the `red` module shows the currently installed version of Red.
+The module will be shown if any of the following conditions are met:
+
+- The current directory contains a file ending with `.red` or `.reds`
```
Comment (vladimyr) [inline docs/config/README.md:2245]: Follow wording style set by other doc sections:
```suggestion
- The current directory contains a file with `.red` or `.reds` extension
```

File: src/modules/red.rs (hunk 2 above)
Comment (vladimyr) [reply src/modules/red.rs:32]: This is still actual 🙃

File: src/modules/red.rs (hunk 3 above)
Comment (vladimyr) [reply src/modules/red.rs:51]: Avoid adding clippy warning overrides - simply return a `String`.

**Review by vladimyr — CHANGES_REQUESTED** [review]

**Comment by vladimyr** [comment]
@Milo123459 Please apply this patch https://github.com/vladimyr/starship/commit/0d7c53c.patch

**Review by vladimyr — APPROVED** [review]
@Milo123459 Thanks for your great work and patient walk through this review journey!

LGTM

**Comment by andytom** [comment]
Thank you @Milo123459 for your contribution and as always thank you for reviewing @vladimyr.

**Pull request merged**